core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -85,8 +86,9 @@ abstract class AbstractFetcherManager(protected val name: String, clientId: Stri
             fetcherThread.start
         }
 
-        fetcherThreadMap(brokerAndFetcherId).addPartitions(partitionAndOffsets.map { case (topicAndPartition, brokerAndInitOffset) =>
-          topicAndPartition -> brokerAndInitOffset.initOffset
+        fetcherThreadMap(brokerAndFetcherId).addPartitions(partitionAndOffsets.map {
+          case (TopicAndPartition(t, p), brokerAndInitOffset) =>
```

Comment:
minor: seems like it would just be a little more work to use `TopicPartition` from the two callers of `addFetcherForPartitions`.